dashboard/test/integration/curriculum_docs_test.rb

```diff
@@ -1,103 +1,165 @@
+        {path: '/docs/gamelab/draw', method: :get}
+      )
+
+      assert_recognizes(
+        {controller: 'programming_expressions', action: 'docs_show', programming_environment_name: 'gamelab', programming_expression_key: 'draw'},
+        {path: '/docs/gamelab/draw/index.html', method: :get}
+      )
+    end
+
+    test '/docs/spritelab/<key> is routed to ProgrammingExpressionsController' do
+      assert_recognizes(
+        {controller: 'programming_expressions', action: 'docs_show', programming_environment_name: 'spritelab', programming_expression_key: 'gamelab_turn'},
+        {path: '/docs/spritelab/gamelab_turn', method: :get}
+      )
+
+      assert_recognizes(
+        {controller: 'programming_expressions', action: 'docs_show', programming_environment_name: 'spritelab', programming_expression_key: 'gamelab_turn'},
+        {path: '/docs/spritelab/gamelab_turn/index.html', method: :get}
+      )
+    end
+
+    test '/docs/weblab/<key> is routed to ProgrammingExpressionsController' do
+      assert_recognizes(
+        {controller: 'programming_expressions', action: 'docs_show', programming_environment_name: 'weblab', programming_expression_key: 'style'},
+        {path: '/docs/weblab/style', method: :get}
+      )
+
+      assert_recognizes(
+        {controller: 'programming_expressions', action: 'docs_show', programming_environment_name: 'weblab', programming_expression_key: 'style'},
+        {path: '/docs/weblab/style/index.html', method: :get}
+      )
+    end
   end
 
-  test '/docs/gamelab is routed to ProgrammingEnvironmentsController' do
-    assert_recognizes(
-      {controller: 'programming_environments', action: 'docs_show', programming_environment_name: 'gamelab'},
-      {path: '/docs/gamelab', method: :get}
-    )
+  class CodeDocsCachingTest < CurriculumDocsTest
+    def setup
+      Script.stubs(:should_cache?).returns true
+      @programming_environment = create :programming_environment
+      programming_environment_category = create :programming_environment_category, programming_environment: @programming_environment
+      @programming_expression = create :programming_expression, programming_environment: @programming_environment, programming_environment_category: programming_environment_category
+    end
+
+    test "environment index should cache all queries" do
+      assert_cached_queries(0) do
+        get programming_environments_path
+      end
+      assert_response :success
+    end
+
+    test "environment show should cache all queries" do
+      assert_cached_queries(0) do
+        get programming_environment_path(@programming_environment.name)
+      end
+      assert_response :success
+    end
+
+    test "expression show should cache all queries" do
+      assert_cached_queries(0) do
+        get programming_environment_programming_expression_path(@programming_environment.name, @programming_expression.key)
+      end
+      assert_response :success
+    end
   end
```

Comment:
apologies again... these were the query count tests I was asking for, which clearly you had already written 🤦  I'm not sure how I missed them! 

it does still seem somewhat valuable to have the new uncached query count tests below though, since if we try to change the caching implementation we'll be able to tell if we are regressing levelbuilder.

Reply:
I only added them yesterday so they're pretty new :) 

> it does still seem somewhat valuable to have the new uncached query count tests below though

I'm not sure what you mean here sorry. Are there more tests I should add?

Reply:
no, these are plenty! 

ok, I think I am just confused about what order everything happened in. when I look at the PR overview, it looks like you had already written the `assert_cached_query` tests before I had written them, so I was apologizing for overlooking them.

Reply:
ok I misunderstood you again. bad day for dave trying to communicate...

> it does still seem somewhat valuable to have the new uncached query count tests below though

No, I am not requesting more tests. the assert_cached_query tests are the ones I meant to request. these assert_query tests also seem valuable, even though I didn't mean to request them -- I only meant to request the assert_cached_query tests.

Reply:
Sounds good! Thanks for looking at this review so carefully! It seems like we're in a good state now :)